light-base/src/sync_service/standalone.rs

```diff
@@ -1223,12 +1274,24 @@ impl<TPlat: PlatformRef> Task<TPlat> {
                             HashDisplay(&verified_hash),
                             error
                         );
+
+                        // TODO: ban peers that have announced the block
```

Comment:
Should be done before merging

Reply:
I've changed my mind about the fact that this is needed before merging.
We obviously need the help of the sync state machine here, but its guarantees in corner cases (such as knowing which sources know about a bad block) are not very well defined. The sync state machine needs a general clean-up, which is currently blocked on #1409.

I don't think that banning peers that advertise bad blocks is critical, and so it's fine to leave it as a todo.